### PR TITLE
Chore: Warnings and Clippy

### PR DESCRIPTION
**Pull request by aphelionz**
This PR fixes various rustc and clippy warnings 

Note: This should follow #19 as it contains those commits as well.

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -269,7 +268,6 @@ impl InstanceFilter<Call> for ProxyType {
 			),
 			ProxyType::Governance => matches!(
 				c,
-				Call::Democracy(..) |
 				Call::Council(..) |
```
Comment (regenaissance) [inline bin/node/runtime/src/lib.rs:271]: We can probably take out council and technical commitee as well

Comment (aphelionz) [reply bin/node/runtime/src/lib.rs:271]: Sounds good. I can do that in a follow-on PR to this one

Comment (aphelionz) [reply bin/node/runtime/src/lib.rs:271]: Actually I noted this in the previous PR - `committee` is tied pretty directly to a lot of things especially when you start running the tests. I'll do this next.

Comment (regenaissance) [reply bin/node/runtime/src/lib.rs:271]: ah ok we can leave it then

File: frame/social-swap2/src/lib.rs [hunk 2]
```diff
@@ -192,10 +192,10 @@ decl_module! {
...
             if amount0_out > 0u32.into() {
-				T::FungibleToken::transfer(&social_token_id, &token0, &to, amount0_out);
+				T::FungibleToken::transfer(&social_token_id, &token0, &to, amount0_out)?;
             }
             if amount1_out > 0u32.into() {
-				T::FungibleToken::transfer(&social_token_id, &token1, &to, amount1_out);
+				T::FungibleToken::transfer(&social_token_id, &token1, &to, amount1_out)?;
             }
             // TODO:
             // if (data.length > 0) IUniswapV2Callee(to).uniswapV2Call(msg.sender, amount0Out, amount1Out, data);
```
Comment (andor0) [inline frame/social-swap2/src/lib.rs]: it can remove because `data.length` equals `0` usually (https://uniswap.org/docs/v2/smart-contracts/pair/#swap-1)

**Review by andor0 — APPROVED** [review]

**Review by andor0 — APPROVED** [review]

**Pull request merged**